Samples/CommandBinding/MauiEdit.ViewModels/Controller/MainFormController.cs

```diff
@@ -0,0 +1,237 @@
+        : base(serviceProvider)
+    {
+        // The initial sample text:
+        TextDocument = $"""
+                This is a test document.
+                It doesn't contain a lot of information.
+                But it contains enough, to test this app.
+
+                Just a few lines of text are sufficient for that.
+
+                Today is {DateTime.Now:dddd, MMMM dd. yyyy}. It is {DateTime.Now: hh:mm tt}.
+
+                The end.
+                """;
+    }
+
+    /// <summary>
+    ///  Provides the Document for the Editor.
+    /// </summary>
+    public string? TextDocument
+    {
+        get => _textDocument;
+
+        set
+        {
+            var wasEmpty = string.IsNullOrEmpty(_textDocument);
+
+            if (SetProperty(ref _textDocument, value))
+            {
+                _lines = GetLinesFromDocument();
+
+                if (wasEmpty ^ string.IsNullOrEmpty(_textDocument))
+                {
+                    NewDocumentCommand.NotifyCanExecuteChanged();
+                    ToUpperCommand.NotifyCanExecuteChanged();
+                }
+            }
+        }
+    }
+
+    // The CarriageReturn string we detect.
+    private string CarriageReturn
+        => GetCarriageReturnStringBasedOnDocument() ?? "\r\n";
+
+    /// <summary>
+    ///  Returns a list of lines of the document.
+    /// </summary>
+    public ReadOnlyMemory<char>[] Lines
+        => _lines;
+
+    // This is an example how to implement the On...Changed method for a property,
+    // which has been automatically generated. The MVVM Community Toolkit provides 
+    // partial On...Methods methods for generated properties which can be amended like this:
+    partial void OnSelectionLinesChanged((int StartLine, int EndLine) value)
+        => Debug.Print($"SelectionLine: {_selectionLines}");
+
+    partial void OnSelectionLengthChanged(int value)
+        => UpdateSelectionInfo();
+
+    partial void OnSelectionIndexChanged(int value)
+        => UpdateSelectionInfo();
+
+    /// <summary>
+    ///  Triggers calculation of SelectionRow, SelectionColumn and SelectionLines.
+    /// </summary>
+    private void UpdateSelectionInfo()
+    {
+        (SelectionRow, SelectionColumn) = GetRowAndColumnFromPosition(_selectionIndex);
+
+        SelectionLines = (
+            SelectionRow,
+            GetRowAndColumnFromPosition(_selectionIndex + _selectionLength).Row);
+    }
+
+    // Finds out, how the line breaks are coded, which unfortunately
+    // is different for the respected textbox controls on different platforms.
+    //
+    // This could be done more reliably, by having a dedicated
+    // Property, which determines the platform's behavior directly
+    // through a platform-depending binding.
+    // But this here works too.
+    private string? GetCarriageReturnStringBasedOnDocument()
+    {
+        if (TextDocument is null)
+        {
+            return null;
+        }
+
+        if (TextDocument?.IndexOf("\r\n") > -1)
```

Comment:
Presumably this would fail for "\n\r\n\r". Wouldn't it be better to look for just "\r" and check before and after?